core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java

```diff
@@ -333,75 +331,57 @@ private boolean setChannel(ChannelXmlElement channelXmlElement, TelemetryConfigu
     private String getConfigurationFile() {
 
         // Trying to load configuration as a resource.
-        ClassLoader classLoader = TelemetryConfigurationFactory.class.getClassLoader();
-        URL resource = classLoader.getResource(fileToParse);
-        if (resource != null) {
-            String configurationFile = resource.getFile();
-            InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "Configuration file has been successfully found in: '%s'", configurationFile);
-            return configurationFile;
-        }
+        String configurationFile = getConfigurationFromDefaultClassLoader();
 
-        InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "Could not find resource named '%s'", fileToParse);
         // If not found as a resource, trying to load from the executing jar directory
-        String configurationFile =  getConfigurationFromLibraryLocation();
-        if (configurationFile != null) {
-            return configurationFile;
+        if (configurationFile == null) {
+            configurationFile =  getConfigurationFromLibraryLocation();
         }
 
-        // If not found as a resource, trying to load from the class path
-        return getConfFromClassPath(classLoader);
-    }
-
-    private String getConfFromClassPath(ClassLoader classLoader) {
-        if (!(classLoader instanceof URLClassLoader)) {
-            return null;
+        // If still not found try to get it from the class path
+        if (configurationFile == null) {
```

Comment:
This should be in first check of the 'if (configurationFile == null)' this is activated only if the second one produces null